### ISSUE TITLE
Integration tests that depend on other modules fail

### ISSUE THREAD

**Issue by johlju** [PowerShell/DscResource.Tests#168]
When building integration tests which depends on other resources in other modules, those fail because they don't exist on the build worker. I could download them prior to running the tests, but thinking if we could re-use the same logic that was introduced for examples in in PR #163.

For this integration test I want the test code to automatically install the dependent modules xStorage and PSDscResources.
xSQLServer Integration test: https://github.com/johlju/xSQLServer/blob/integration-test-xSQLServerSetup/Tests/Integration/MSFT_xSQLServerSetup.config.ps1

**Comment by PlagueHO** [comment]
We could move the code that optionally installs or warns the user to a new function so that it can be resused. Shouldn't be too difficult.

**Comment by johlju** [comment]
I will see if I can get this done

**Comment by johlju** [comment]
I have moved the code to a separate helper function (also found a bug with 'Version' that I fixed, see [line 1213](https://github.com/johlju/DscResource.Tests/blob/automatic-install-tests-dependencies/TestHelper.psm1#L1213)).

Looking how the Pester tests are run, my thought is to do the following.

* Before calling Invoke-Pester. Loop thru all the *.config.ps1 files (`Get-ChildItem -Path '*.config.ps1' -Recurse`) and call `Get-ResourceModulesInConfiguration` and `Install-DependentModule` for each config file. 

My thought is to add this before the evaluation of model type ('Default' or 'Harness'), before [line 204](https://github.com/PowerShell/DscResource.Tests/blob/dev/AppVeyor.psm1#L204). Let me know if there is any issues with that.
